overlay-debs/linux-rb1/linux_6.15.3-1qcom1.debdiff

```diff
@@ -0,0 +1,137 @@
++    CONFIG_QCOM_CLK_SMD_RPM=y, CONFIG_QCOM_APCS_IPC=y.
++  * Set CONFIG_CLK_QCM2290_GPUCC=y, CONFIG_RPMSG_QCOM_SMD=y,
++    CONFIG_QCOM_SMEM=y, CONFIG_QCOM_SMD_RPM=y,
++    CONFIG_INTERCONNECT_QCOM_SMD_RPM=y
++
++ -- Loic Minier <loic.minier@oss.qualcomm.com>  Mon, 23 Jun 2025 14:29:32 +0000
++
+ linux (6.15.3-1~exp1) experimental; urgency=medium
+ 
+   * New upstream stable update:
+diff -Nru linux-6.15.3/debian/config/arm64/config linux-6.15.3/debian/config/arm64/config
+--- linux-6.15.3/debian/config/arm64/config	2025-06-19 16:10:05.000000000 +0000
++++ linux-6.15.3/debian/config/arm64/config	2025-06-23 14:29:32.000000000 +0000
+@@ -187,6 +187,7 @@
+ CONFIG_COMMON_CLK_SCPI=m
+ CONFIG_COMMON_CLK_SI5341=y
+ CONFIG_COMMON_CLK_FSL_SAI=y
++CONFIG_CLK_QCM2290_GPUCC=y
+ CONFIG_CLK_QORIQ=y
+ CONFIG_CLK_LS1028A_PLLDIG=y
+ CONFIG_COMMON_CLK_XGENE=y
+@@ -364,7 +365,7 @@
+ CONFIG_QCOM_CLK_APCS_MSM8916=m
+ CONFIG_QCOM_CLK_APCC_MSM8996=m
+ CONFIG_QCOM_CLK_RPM=m
+-CONFIG_QCOM_CLK_SMD_RPM=m
++CONFIG_QCOM_CLK_SMD_RPM=y
+ CONFIG_QCOM_CLK_RPMH=y
+ CONFIG_MSM_GCC_8916=y
+ CONFIG_MSM_GCC_8996=y
+@@ -372,7 +373,7 @@
+ CONFIG_MSM_GCC_8998=y
+ CONFIG_MSM_GPUCC_8998=m
+ CONFIG_MSM_MMCC_8998=m
+-CONFIG_QCM_GCC_2290=m
++CONFIG_QCM_GCC_2290=y
+ CONFIG_QCM_DISPCC_2290=m
+ CONFIG_SC_CAMCC_7180=m
+ CONFIG_SC_CAMCC_7280=m
+@@ -1031,7 +1032,9 @@
+ CONFIG_INTERCONNECT_QCOM_SC8280XP=m
+ CONFIG_INTERCONNECT_QCOM_SDM845=y
+ CONFIG_INTERCONNECT_QCOM_SM6115=y
++CONFIG_INTERCONNECT_QCOM_SM8150=y
+ CONFIG_INTERCONNECT_QCOM_SM8250=y
++CONFIG_INTERCONNECT_QCOM_SMD_RPM=y
+ CONFIG_INTERCONNECT_QCOM_X1E80100=y
+ 
+ ##
+@@ -1088,7 +1091,7 @@
+ CONFIG_ROCKCHIP_MBOX=y
+ CONFIG_BCM2835_MBOX=y
+ CONFIG_HI6220_MBOX=y
+-CONFIG_QCOM_APCS_IPC=m
++CONFIG_QCOM_APCS_IPC=y
+ CONFIG_XGENE_SLIMPRO_MBOX=m
+ CONFIG_MTK_ADSP_MBOX=m
+ CONFIG_MTK_CMDQ_MBOX=m
+@@ -1914,13 +1917,14 @@
+ CONFIG_PINCTRL_MSM8916=y
+ CONFIG_PINCTRL_MSM8996=y
+ CONFIG_PINCTRL_MSM8998=y
+-CONFIG_PINCTRL_QCM2290=m
++CONFIG_PINCTRL_QCM2290=y
+ CONFIG_PINCTRL_SC7180=y
+ CONFIG_PINCTRL_SC7280=y
+ CONFIG_PINCTRL_SC8180X=y
+ CONFIG_PINCTRL_SC8280XP=y
+ CONFIG_PINCTRL_SDM845=y
+ CONFIG_PINCTRL_SM6115=m
++CONFIG_PINCTRL_SM8150=y
+ CONFIG_PINCTRL_SM8250=m
+ CONFIG_PINCTRL_X1E80100=m
+ 
+@@ -2044,6 +2048,7 @@
+ CONFIG_REGULATOR_PFUZE100=m
+ CONFIG_REGULATOR_PWM=m
+ CONFIG_REGULATOR_QCOM_PM8008=m
++CONFIG_REGULATOR_QCOM_REFGEN=m
+ CONFIG_REGULATOR_QCOM_RPM=m
+ CONFIG_REGULATOR_QCOM_RPMH=m
+ CONFIG_REGULATOR_QCOM_SMD_RPM=m
+@@ -2065,6 +2070,7 @@
+ CONFIG_QCOM_Q6V5_PAS=m
+ CONFIG_QCOM_Q6V5_WCSS=m
+ CONFIG_QCOM_SYSMON=m
++CONFIG_QCOM_TZMEM_MODE_SHMBRIDGE=y
+ CONFIG_QCOM_WCNSS_PIL=m
+ CONFIG_TI_K3_R5_REMOTEPROC=m
+ 
+@@ -2082,10 +2088,13 @@
+ ##
+ ## file: drivers/rpmsg/Kconfig
+ ##
++CONFIG_RPMSG_CHAR=m
++CONFIG_RPMSG_CTRL=m
++CONFIG_RPMSG_VIRTIO=y
+ CONFIG_RPMSG_MTK_SCP=m
+ CONFIG_RPMSG_QCOM_GLINK_RPM=m
+ CONFIG_RPMSG_QCOM_GLINK_SMEM=m
+-CONFIG_RPMSG_QCOM_SMD=m
++CONFIG_RPMSG_QCOM_SMD=y
```

Comment:
This is obsolete and can go away completely.

Reply:
I tried dropping it, I saw that CONFIG_QCOM_SMD_RPM depends on this being =y or =n. The d-i configs of the Debian kernel expect the module though – "missing module qcom_smd", so I wonder if it's still useful on some platforms.

I guess I will want to upstream all these changes to Debian and will need your help to backup things that are considered obsolete. 

This one is set in the upstream defconfig too.

Reply:
No, corresponding DTs were changes before 6.6. We can drop the module from d-i config too.

Reply:
I don't understand your remark; are you saying upstream linux stopped caring about platforms that would benefit from this config since 6.6? Do you agree we should drop it in the upstream defconfig first?

Reply:
I did have to change this config to =y for the kernel to boot on RB1; it could be bad interactions with other configs, and perhaps things would work if this was off, happy to try something else to help fix the upstream defconfig and the Debian one. :)

Reply:
Ok, after some config dependencies digging, I resolved this by dropping CONFIG_QCOM_SMD_RPM and setting CONFIG_RPMSG_QCOM_GLINK_SMEM=y